### DISCUSSION TITLE
[CONTRACTS] - uAD Protocol Integration Bounty

### DISCUSSION THREAD

**Discussion by 0x4007** [pavlovcik/uad-ui#129] [Ideas]
Leveraging transfer hooks...

1. We tell defi protocols to add our uAD as a supported asset. 
2. Add a whitelist entry of their deposit address
3. Add custom logic for them to collect a fee or some incentive, for example; they capture 1% of the uAD deposited and it goes into their treasury. 

If we write that transfer hook it could be an extremely low friction way to have bounties for our dollar. 

We could also pay a transaction fee in debt tokens or governance tokens depending on the integration partner.

**Comment by Draeieg** [comment]
We could stack the yields other projects/protocols have for their governance tokens or ay other stacking mechanism they have using the proxy yield aggregator

then we make a list of "partner" projects using the same mechanism we have for UBQ so that they get a discount on the upfront fee if they use the governance token of their project

so that we don't hurt UBQ by doing that we could add some psyops the the UI so that it says "which token would you like to match" and we show UBQ, "others"  or nothing, so when staking is easier to use our own but still possible to user others after sorting through a (in the future) big list

also the matching requirements to get discounts could be higher for other projects vs using UBQ,  either making the UBQ amount lower or the partners projects higher

**Comment by 0x4007** [comment]
> Leveraging transfer hooks...
> 
> 1. We tell defi protocols to add our uAD as a supported asset.
> 2. Add a whitelist entry of their deposit address
> 3. Add custom logic for them to collect a fee or some incentive, for example; they capture 1% of the uAD deposited and it goes into their treasury.
> 
> If we write that transfer hook it could be an extremely low friction way to have bounties for our dollar.
> 
> We could also pay a transaction fee in debt tokens or governance tokens depending on the integration partner.

Discussion from 11 October - @hpiyankov suggests to create a "pot" of UBQ and distribute from the daily pot; otherwise a "symbolic" amount (low amount)
